resource/config.go

```diff
@@ -24,6 +24,7 @@ type Config struct {
 	AssociatedResourceConfigs []AssociatedResourceConfig
 	Attributes                utils.AttributeMap
 
+	DiffingAttributes   ConfigValidator
```

Comment:
Thanks for all the discussion offline! On further reflection, I'm confused why `DiffingAttributes` is needed at all. Can we just nil out `ConvertedAttributes` in the `Equals` method? Are there possible salient semantic differences between resource configs with differing converted attributes? An attribute map converter can only be specified at the time of resource registration, so I don't see how a user could make a change to a config that results in _only_ converted attributes changing.